vividus-plugin-azure-service-bus/src/main/java/org/vividus/azure/servicebus/service/ServiceBusService.java

```diff
@@ -17,57 +17,153 @@
+import org.vividus.util.property.PropertyMappedCollection;
 
 public class ServiceBusService
 {
     private static final Logger LOGGER = LoggerFactory.getLogger(ServiceBusService.class);
 
     private static final int CLIENT_MAX_RETRIES = 1;
-    private static final Duration CLIENT_TRY_TIMEOUT = Duration.ofSeconds(1);
+    private static final Duration CLIENT_TRY_TIMEOUT = Duration.ofSeconds(10);
+    private static final Class<?> MESSAGES_KEY = ServiceBusReceivedMessage.class;
+    private static final String DEFAULT_NAMESPACE_POSTFIX = ".servicebus.windows.net";
 
     private ServiceBusClientBuilder clientBuilder;
+    private Map<String, ServiceBusProcessorClient> clients = new HashMap<>();
+    private TestContext testContext;
+    private PropertyMappedCollection<ServiceBusClientEntity> clientConfigs;
 
-    public ServiceBusService(TokenCredential tokenCredential)
+    public ServiceBusService(TokenCredential tokenCredential, TestContext testContext,
+                             PropertyMappedCollection<ServiceBusClientEntity> clientConfigs)
     {
         this.clientBuilder = new ServiceBusClientBuilder()
                 .credential(tokenCredential)
                 .retryOptions(new AmqpRetryOptions().setMaxRetries(CLIENT_MAX_RETRIES)
                         .setTryTimeout(CLIENT_TRY_TIMEOUT));
+        this.testContext = testContext;
+        this.clientConfigs = clientConfigs;
     }
 
-    public void send(ChannelType type, String name, String namespaceName, String message)
+    public void send(String clientKey, String message)
     {
+        ServiceBusClientEntity serviceBusClientEntity = getServiceBusClientConfig(clientKey);
+
+        String namespace = serviceBusClientEntity.getNamespace();
         ServiceBusClientBuilder.ServiceBusSenderClientBuilder senderBuilder = clientBuilder
-                .fullyQualifiedNamespace(namespaceName + ".servicebus.windows.net")
+                .fullyQualifiedNamespace(namespace + DEFAULT_NAMESPACE_POSTFIX)
                 .sender();
-        if (type == ChannelType.QUEUE)
+
+        ChannelType channelType = serviceBusClientEntity.getChannelType();
+        String name = serviceBusClientEntity.getName();
+        if (channelType == ChannelType.QUEUE)
         {
             senderBuilder.queueName(name);
         }
         else
         {
             senderBuilder.topicName(name);
         }
+
         try (ServiceBusSenderClient client = senderBuilder.buildClient())
         {
             client.sendMessage(new ServiceBusMessage(message));
-            LOGGER.info("The message was successfully sent to {} in {} namespace", name, namespaceName);
+            LOGGER.info("The message was successfully sent to {} in {} namespace", name, namespace);
         }
         catch (ServiceBusException e)
         {
             throw new ServiceBusSendMessageException("Unable to send message to Azure Service Bus", e);
         }
     }
+
+    public void startConsuming(String clientKey)
```

Comment:
do we need to handle cases when end user invokes startConsuming several times with the same key?